### ISSUE TITLE
Adaptation to Mysql

### ISSUE THREAD

**Issue by amxyz-cyber** [jonschoning/espial#41]
Dear Developer of Espial,

I could successfully migrate espial from sqlite to mysql. However, there is one thing that simply doesn't work. The search realized in the function **Model.bookmarksTagsQuery** does not produce any search results. It's because I can't create a customized group_concat function for mysql. THe mysql version contains no commas:
https://www.mysqltutorial.org/mysql-group_concat/

Could you please give me a workable alternative to **Model.sqliteGroupConcat**. As you implemented such a complicated search I assume that you can implement a group_concat version for mysql. Here is an example of mysql's group_concat:

SELECT  GROUP_CONCAT(DISTINCT country ORDER BY country SEPARATOR ' ')
FROM customers;

**Comment by jonschoning** [comment]
I haven't tested if this works at all, but my first guess would be something like this

```haskell
import Database.Esqueleto.Internal.Internal (unsafeSqlFunction, SqlExpr(..), noMeta, valueToFunctionArg)

mysqlGroupConcat ::
     PersistField a
  => SqlExpr (Value a)
  -> SqlExpr (Value a)
  -> SqlExpr (Value Text)
mysqlGroupConcat expr sep =
  ERaw noMeta $ \_ info ->
    let (tlbExpr, _) = valueToFunctionArg info expr
        (tlbSep, valSep) = valueToFunctionArg info sep
    in ("GROUP_CONCAT(" <> tlbExpr <> " SEPARATOR '" <> tlbSep <> "')", valSep)
```

**Comment by amxyz-cyber** [comment]
Hi,
many thanks for your help. I had to change the code a bit. It compiled successfully. But when starting the web server, mysql issued an error message because of a wrong syntax. This is the code that worked for me:

`mysqlGroupConcat ::
       PersistField a
  => SqlExpr (Value a)
  -> SqlExpr (Value Text)
mysqlGroupConcat expr =
  ERaw noMeta $ \_ info ->
    let (tlbExpr, yx) = valueToFunctionArg info expr
    in ("GROUP_CONCAT(" <> tlbExpr <> " SEPARATOR ' ' )", yx)`

I also found out why your query doesn't yield any results. It's because of the like-operator. But when performing a simple query of your database using the Mysql workbench it works. So I had to use the match against function for fulltext search in mysql. As for the match against function your code was really helpful. And now it's working.
